src/NServiceBus.RavenDB/Outbox/RavenDbOutboxStorage.cs

```diff
@@ -21,42 +20,52 @@ protected override void Setup(FeatureConfigurationContext context)
             DocumentStoreManager.GetUninitializedDocumentStore<StorageType.Outbox>(context.Settings)
                 .CreateIndexOnInitialization(new OutboxRecordsIndex());
 
-            context.Container.ConfigureComponent(b =>
-            {
-                return new OutboxPersister(endpointName, b.Build<IOpenTenantAwareRavenSessions>());
-            }, DependencyLifecycle.InstancePerCall);
+            context.Container.ConfigureComponent(b => new OutboxPersister(endpointName, b.Build<IOpenTenantAwareRavenSessions>()), DependencyLifecycle.InstancePerCall);
 
-            context.Container.ConfigureComponent(b =>
-            {
-                var store = DocumentStoreManager.GetDocumentStore<StorageType.Outbox>(context.Settings, b);
-                return new OutboxRecordsCleaner(store);
-            }, DependencyLifecycle.InstancePerCall);
+            var frequencyToRunDeduplicationDataCleanup = context.Settings.GetOrDefault<TimeSpan?>(FrequencyToRunDeduplicationDataCleanup) ?? TimeSpan.FromMinutes(1);
+            var timeToKeepDeduplicationData = context.Settings.GetOrDefault<TimeSpan?>(TimeToKeepDeduplicationData) ?? TimeSpan.FromDays(7);
+            var cleanupDisabled = frequencyToRunDeduplicationDataCleanup == Timeout.InfiniteTimeSpan;
 
-            context.Container.ConfigureComponent<OutboxCleaner>(DependencyLifecycle.InstancePerCall);
+            if (!cleanupDisabled)
+            {
+                context.Settings.AddStartupDiagnosticsSection(StartupDiagnosticsSectionName, new
+                {
+                    CleanupDisabled = false,
+                    TimeToKeepDeduplicationData = timeToKeepDeduplicationData,
+                    FrequencyToRunDeduplicationDataCleanup = frequencyToRunDeduplicationDataCleanup
+                });
 
-            context.RegisterStartupTask(builder => builder.Build<OutboxCleaner>());
+                context.RegisterStartupTask(builder =>
+                {
+                    var store = DocumentStoreManager.GetDocumentStore<StorageType.Outbox>(context.Settings, builder);
+                    return new OutboxCleaner(new OutboxRecordsCleaner(store), frequencyToRunDeduplicationDataCleanup, timeToKeepDeduplicationData);
+                });
+            }
+            else
+            {
+                context.Settings.AddStartupDiagnosticsSection(StartupDiagnosticsSectionName, new
+                {
+                    CleanupDisabled = true,
```

Comment:
Using the same style as SqlPersistence of not showing the settings when they are not used. When the cleanup is disabled we will never use `TimeToKeepDeduplicationData` or `FrequencyToRunDeduplicationDataCleanup` therefore there is no point in adding it to the section.